fragdenstaat_de/fds_cms/static/js/foirequest_map.js

```diff
@@ -0,0 +1,112 @@
+
+// Cache for status icons
+const statusIconCache = {};
+
+function getStatusIcon(status) {
+    if (statusIconCache[status]) {
+        return statusIconCache[status];
+    }
+
+    const statusColors = {
+        'pending': '#ffc107',
+        'resolved': '#28a745',
+        'refused': '#dc3545',
+        'partially_successful': '#17a2b8',
+        'overdue': '#fd7e14',
+        'asleep': '#6c757d'
+    };
+
+    const color = statusColors[status] || '#6c757d';
+    const icon = L.divIcon({
+        className: 'custom-div-icon',
+        html: `<div style="background-color: ${color}; width: 12px; height: 12px; border-radius: 50%; border: 2px solid white; box-shadow: 0 0 4px rgba(0,0,0,0.3);"></div>`,
+        iconSize: [16, 16],
+        iconAnchor: [8, 8]
+    });
```

Comment:
color should not be the only way to distinguish between the different result statuses https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html